spec/rubocop/cop/rails/transaction_requires_new_spec.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::Rails::TransactionRequiresNew do
+  subject(:cop) { described_class.new(config) }
+
+  let(:config) { RuboCop::Config.new }
+
+  context 'when no arguments are passed' do
+    it 'registers an offense when using `ActiveRecord::Base#transaction`' do
+      expect_offense(<<~RUBY)
+        ActiveRecord::Base.transaction do
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Always pass "requires_new: true" to transactions.
+        end
+      RUBY
+
+      expect_correction(<<~RUBY)
+        ActiveRecord::Base.transaction(requires_new: true) do
+        end
+      RUBY
+    end
+
+    it 'registers an offense when using a Model transaction' do
+      expect_offense(<<~RUBY)
+        User.transaction do
+        ^^^^^^^^^^^^^^^^^^^ Always pass "requires_new: true" to transactions.
+        end
+      RUBY
+
+      expect_correction(<<~RUBY)
+        User.transaction(requires_new: true) do
+        end
+      RUBY
+    end
+  end
+
+  context 'when arguments other than `requires_new: true` are passsed' do
+    it 'registers an offense when using `ActiveRecord::Base#transaction`' do
+      expect_offense(<<~RUBY)
+        ActiveRecord::Base.transaction(some_option: false) do
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Always pass "requires_new: true" to transactions.
+        end
+      RUBY
+
+      # Does NOT auto-correct when arugments are present
+      expect_correction(<<~RUBY)
+        ActiveRecord::Base.transaction(some_option: false) do
```

Comment:
I think it can be auto-corrected for this offense.

Reply:
Yes, I suppose it can be.  I imagine that it will be safe enough to simply add `requires_new: true` as the last argument in a transaction.